src/main/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReference.java

```diff
@@ -79,45 +77,46 @@ public J visitLambda(J.Lambda lambda, ExecutionContext executionContext) {
-                                body = tree;
-                                code = "#{}.class::cast";
+                        J tree = j.getTree();
+                        if ((tree instanceof J.Identifier || tree instanceof J.FieldAccess) &&
+                            !(j.getType() instanceof JavaType.GenericTypeVariable)) {
+                            J.FieldAccess classLiteral = newClassLiteral(((Expression) tree).getType(), tree instanceof J.FieldAccess);
+                            if (classLiteral != null) {
+                                //noinspection DataFlowIssue
+                                JavaType.FullyQualified classType = ((JavaType.Parameterized) classLiteral.getType()).getType();
+                                Optional<JavaType.Method> castMethod = classType.getMethods().stream().filter(m -> m.getName().equals("cast")).findFirst();
+                                if (castMethod.isPresent()) {
+                                    return newInstanceMethodReference(castMethod.get(), classLiteral, lambda.getType()).withPrefix(lambda.getPrefix());
+                                }
                             }
                         }
                     }
                 }
 
-                if (body instanceof J.Identifier && !code.isEmpty()) {
-                    J.Identifier identifier = (J.Identifier) body;
-                    JavaType.FullyQualified fullyQualified = TypeUtils.asFullyQualified(identifier.getType());
-                    @Language("java") String stub = fullyQualified == null ? "" :
-                            "package " + fullyQualified.getPackageName() + "; public class " +
-                            fullyQualified.getClassName();
-                    JavaTemplate template = JavaTemplate.builder(code)
-                            .contextSensitive()
-                            .javaParser(JavaParser.fromJavaVersion().dependsOn(stub))
-                            .imports(fullyQualified == null ? "" : fullyQualified.getFullyQualifiedName()).build();
-                    return template.apply(getCursor(), l.getCoordinates().replace(), identifier.getSimpleName());
-                } else if (body instanceof J.Binary) {
+                if (body instanceof J.Binary) {
                     J.Binary binary = (J.Binary) body;
                     if (isNullCheck(binary.getLeft(), binary.getRight()) ||
                         isNullCheck(binary.getRight(), binary.getLeft())) {
-                        maybeAddImport("java.util.Objects");
-                        code = J.Binary.Type.Equal.equals(binary.getOperator()) ? "Objects::isNull" :
-                                "Objects::nonNull";
+                        doAfterVisit(new ShortenFullyQualifiedTypeReferences().getVisitor());
+                        code = J.Binary.Type.Equal.equals(binary.getOperator()) ? "java.util.Objects::isNull" :
+                                "java.util.Objects::nonNull";
```

Comment:
OpenRewrite doesn't provide any API to do a lookup of types. That would sometimes be useful to have.